File: JavaScript/7kyu/SumOfNumbers.js
function getSum(a, b) {
  var x = 0;
  for (let i = Math.min(a, b); i < Math.max(a, b); i++) {
    x += i;
  }
  return x;
}

function getSum_2(a, b) {
  let min = Math.min(a, b);
  let max = Math.max(a, b);

  return ((max - min + 1) * (min + max)) / 2;
}

function getSum_3(a, b) {
  return ((a + b) * (Math.abs(a - b) + 1)) / 2;
}

console.log(getSum_3(0, -1));

File: Python/7kyu/SumOfNumbers.py
import sys


def get_sum(a, b):
    tab = sorted([a, b])

    return sum(x for x in range(tab[0], tab[1]+1))


def get_sum_2(a, b):
    return sum(range(min(a, b), max(a, b)))


def get_sum_3(a, b):
    return (a+b) * (abs(a-b) + 1) // 2  # floor division


def firstn(n):
    num = 0
    while num < n:
        yield num
        num += 1
